### PR TITLE
Adding QCBMAnsatz class. Task-templates and workflows need updates

### PR DESCRIPTION
**Pull request by simonwa7**

### CODE REVIEW

File: src/python/zquantum/qcbm/ansatz_utils.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
+from zquantum.core.circuit import Circuit, Gate, Qubit
+import numpy as np
+from typing import List
+
+
+def get_single_qubit_layer(
```
Comment (mstechly) [inline src/python/zquantum/qcbm/ansatz_utils.py:6]: We already have a method for that in `core.circuit.utils` AFAIR.

File: src/python/zquantum/qcbm/cost_function_test.py [hunk 2]
```diff
@@ -65,44 +74,18 @@ def test_init(self):
...
     def test_evaluate(self):
         # Given
-        num_qubits = 4
-        topology = "all"
         epsilon = 1e-6
-        n_layers = 1
-        params = np.array([0, 0, 0, 0])
-        distance_measure = compute_clipped_negative_log_likelihood
-        ansatz = get_qcbm_ansatz(num_qubits, n_layers, topology)
-        target_bitstring_distribution = BitstringDistribution(
```
Comment (mstechly) [inline src/python/zquantum/qcbm/cost_function_test.py:75]: Just curious – why did you get rid of all this?

Comment (simonwa7) [reply src/python/zquantum/qcbm/cost_function_test.py:75]: It's now declared in the `setUp` for the unittest class

File: src/python/zquantum/qcbm/ansatz_utils.py (hunk 1 above)
Comment (simonwa7) [reply src/python/zquantum/qcbm/ansatz_utils.py:6]: Just seeing `create_layer_of_gates` now, but it looks as if it doesn't handle parameterized gates. We should probably go and add optional support there for parameterized gates. Not sure if this is something that should be done now or wait for later?

Comment (simonwa7) [reply src/python/zquantum/qcbm/ansatz_utils.py]: Addressed here: https://github.com/zapatacomputing/z-quantum-core/pull/54/files and in this commit: f546e83

**Pull request merged**